src/lib/device/device.h

```diff
@@ -95,10 +96,11 @@ ENUM(chunk_selector_t, uint8_t, // What are the options to select chunk to alloc
 struct vdev_parameters {
     std::string vdev_name;                  // Name of the vdev
     uint64_t vdev_size;                     // Current Vdev size.
-    uint32_t num_chunks;                    // Total number of primary chunks.
+    uint32_t num_chunks{};                  // Total number of primary chunks.
                                             // NOTE: If pdev opts is ALL_PDEV_STRIPED, then num_chunks would round off
                                             // to number of pdevs evenly
     uint32_t blk_size;                      // Block size vdev operates on
+    uint32_t chunk_size{};                  // Chunk size provided for dynamic vdev.
```

Comment:
If user has to define either chunk_size or num_chunks, should we put them as std::variant?

Reply:
As discussed added vdev_size_type_t enum {STATIC, DYNAMIC} in format param. If it static it takes size and num_chunks or chunk_size. If its dynamic,  it needs only chunk_size.